File: Plugins/Orimath.UITest/ViewModels/AlignmentConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace Orimath.UITest.ViewModels
{
    [ValueConversion(typeof(bool), typeof(VerticalAlignment))]
    public class VerticalAlignmentConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) => 
            value is bool @bool && @bool
                ? VerticalAlignment.Stretch
                : VerticalAlignment.Center;

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) => throw new NotImplementedException();
    }

    [ValueConversion(typeof(bool), typeof(HorizontalAlignment))]
    public class HorizontalAlignmentConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) =>
            value is bool @bool && @bool
                ? HorizontalAlignment.Stretch
                : HorizontalAlignment.Center;

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) => throw new NotImplementedException();
    }
}

File: Plugins/Orimath.UITest/ViewModels/AlignmentConverter.fs
namespace Orimath.UITest.ViewModels
open System
open System.Windows
open System.Windows.Data
open System.Windows.Media

[<ValueConversion(typeof<bool>, typeof<VerticalAlignment>)>]
type VerticalAlignmentConverter() =
    interface IValueConverter with
        member _.ConvertBack(_, _, _, _) = raise (new NotImplementedException())

        member _.Convert(value, _, _, _) =
            match value with
            | :? bool as b when b -> box VerticalAlignment.Stretch
            | _ -> box VerticalAlignment.Center


[<ValueConversion(typeof<bool>, typeof<HorizontalAlignment>)>]
type HorizontalAlignmentConverter() =
    interface IValueConverter with
        member _.ConvertBack(_, _, _, _) = raise (new NotImplementedException())

        member _.Convert(value, _, _, _) =
            match value with
            | :? bool as b when b -> box HorizontalAlignment.Stretch
            | _ -> box HorizontalAlignment.Center


[<ValueConversion(typeof<string>, typeof<Brush>)>]
type SolidColorBrushConverter() =
    let brushConverter = BrushConverter()
    interface IValueConverter with
        member _.ConvertBack(_, _, _, _) = raise (new NotImplementedException())

        member _.Convert(value, _, _, _) =
            match value with
            | :? string as str ->
                try brushConverter.ConvertFromInvariantString(str)
                with _ -> box Brushes.Black
            | _ -> box Brushes.Black
